hack/lib/catalogsource.bash

```diff
@@ -67,12 +72,13 @@ function install_catalogsource {
     cp "${rootdir}/${index_dorkerfile_path}" "${rootdir}/_output/bkp.Dockerfile"
 
     # Replace bundle reference with previously built bundle
-    bundle="registry.ci.openshift.org/knative/release-${CURRENT_VERSION}:serverless-bundle"
+    bundle="${SERVERLESS_BUNDLE%:*}" # Remove the tag from the match
+    bundle="${SERVERLESS_BUNDLE%@*}" # Remove the sha from the match
     if ! grep "${bundle}" "${rootdir}/${index_dorkerfile_path}"; then
       logger.error "Bundle ${bundle} not found in Dockerfile."
       return 1
     fi
-    sed -i "s_\(.*\)\(${bundle}\)\(.*\)_\1image-registry.openshift-image-registry.svc:5000/$OLM_NAMESPACE/serverless-bundle:latest\3_" "${rootdir}/${index_dorkerfile_path}"
+    sed -ri "s#(.*)(${bundle})(:[a-z0-9]*)?(@sha[0-9]+:[a-z0-9]+)?(.*)#\1image-registry.openshift-image-registry.svc:5000/${OLM_NAMESPACE}/serverless-bundle:latest\5#" "${rootdir}/${index_dorkerfile_path}"
```

Comment:
Tested with:

```shell
$ bundle="quay.io/redhat-user-workloads/ocp-serverless-tenant/serverless-operator-release-134/serverless-bundle"
$ sed -r "s#(.*)(${bundle})(:[a-z0-9]*)?(@sha[0-9]+:[a-z0-9]+)?(.*)#\1image-registry.openshift-image-registry.svc:5000/<namespace>/serverless-bundle:latest\5#" olm-catalog/serverless-operator/index/Dockerfile
FROM registry.ci.openshift.org/origin/4.15:operator-registry AS opm

FROM registry.access.redhat.com/ubi9/ubi-minimal as builder

COPY --from=opm /bin/opm /bin/opm

COPY olm-catalog/serverless-operator/index/configs /configs

RUN /bin/opm init serverless-operator --default-channel=stable --output yaml >> /configs/index.yaml
RUN /bin/opm render --skip-tls-verify -o yaml \
quay.io/openshift-knative/serverless-bundle:release-1.32.0 \
registry.ci.openshift.org/knative/release-1.33.0:serverless-bundle \
      registry.ci.openshift.org/knative/release-1.34.0:serverless-bundle >> /configs/index.yaml || \
    /bin/opm render --skip-tls-verify -o yaml \
quay.io/openshift-knative/serverless-bundle:release-1.32.0 \
registry.ci.openshift.org/knative/release-1.33.0:serverless-bundle \
      image-registry.openshift-image-registry.svc:5000/<namespace>/serverless-bundle:latest >> /configs/index.yaml

FROM registry.ci.openshift.org/origin/4.15:operator-registry

COPY --from=builder /configs /configs

LABEL operators.operatorframework.io.index.configs.v1=/configs

ENTRYPOINT ["/bin/opm"]
CMD ["serve", "/configs"]
```

Reply:
The idea is for matching:
1. `(.*)` match starting spaces and anything else until the next match
1. `(${bundle})` match the bundle image without tag or sha
1. `(:[a-z0-9]*)?`  optionally match tag
1. `(@sha[0-9]+:[a-z0-9]+)?`  optionally match sha
1. `(.*)` match the rest

Replace:

1. `\1` Use everything matched by 1
2. `image-registry.openshift-image-registry.svc:5000/${OLM_NAMESPACE}/serverless-bundle:latest` replace the image
3. `\5` use everything matched by 5

Reply:
![image](https://github.com/user-attachments/assets/6f3067d9-262e-4f72-a348-1233716cf7b5)